akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala

```diff
@@ -270,6 +270,7 @@ private[persistence] trait Eventsourced
   /** INTERNAL API. */
   override protected[akka] def aroundPreRestart(reason: Throwable, message: Option[Any]): Unit = {
     try {
+      if (recoveryRunning) message.foreach(self.forward(_))
```

Comment:
> If that behaviour could be added to Eventsourced we wouldn't need the special handling.

message ordering is an important property that we will not relax

Shall we close this idea since it seems difficult and you already have a workaround?

Reply:
Since it is more difficult to do it right than expected we would close it for now. In case we can justify to invest the required amount of time at some point we will reopen it. At least we have a better understanding of the issues now.

Thanks for the feedback.